main.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  # each
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
```

Comment:
- [ ]  `my_each` should return an Enumerator if no block is given:
`p [1, 2, 3].my_each` returns `#<Enumerator>`
This can be achieved with `to_enum`. You can check this behavior in the original `each` method